### PR TITLE
Test scripts tweaks

### PR DESCRIPTION
**Pull request by john-h-kastner**
* The test directory for zlib is not converted, so it isn't expected to compile with the converted library code.
* Similar fix for libarchive
* Undo renames before compiling lua so that the executable links

### CODE REVIEW

File: generate-workflow.py
Comment (mattmccutchen-cci) [inline generate-workflow.py:158]:
```suggestion
        bear make zlib
```
Let's be consistent (and make the build a little faster).  This will have the additional effect that the compilation database will only include the `zlib` target, so you might be able to remove the `--skip` option, but I'll leave it up to you whether you want to try that now or leave it for later.  (Perhaps in general, if we pick the right target, we should never need a `--skip` option?)

**Review by mattmccutchen-cci — CHANGES_REQUESTED** [review]
With these changes and a reasonable-looking workflow run, I'll approve.  Thanks.

File: generate-workflow.py
Comment (john-h-kastner) [reply generate-workflow.py]: I was planning to take a look at all the `--skip` options once we merge the PR for handling multiple definitions of main.

**Comment by john-h-kastner** [comment]
> With these changes and a reasonable-looking workflow run, I'll approve. Thanks.

Run looks good

https://github.com/correctcomputation/actions/actions/runs/687594105

**Review by mattmccutchen-cci — APPROVED** [review]
Hey, it looks like you might have fixed the last error in Lua!  (Unless we find out later that we weren't testing it properly.)  Congratulations!  And thanks again for fixing libarchive and zlib.

**Pull request merged**
